components/diagnostics/diagnose_me/component.yaml

```diff
@@ -23,6 +23,9 @@ description: |-
         Raises:
             RuntimeError: If configuration is not setup properly and
             HALT_ON_ERROR flag is set.
+metadata:
+  annotations:
+    volatile_component: "true"
```

Comment:
Can we add a default cache period like P0D to components? Will that be equivalent to this?

Reply:
>Can we add a default cache period like P0D to components? Will that be equivalent to this?
Yes, it will currently work. But I'm not sure we should use that for several reasons:

1. The fact that component annotations are translated to pod annotations is an implementation detail and a little bit of a design mistake (the components should be portable with no Kubernetes connections).
1. Generally, I do not think that the components should customize their own caching behavior (just like customizing their requirements like CPU, etc), because only the pipeline author knows the right values. But the ability for the components to tell the system that they're volatile/non-deterministic makes sense. The system can then do something with that information. For example, it could ignore it or disable caching or set max cache staleness to 10 minutes. That decision is not made by the component.

Reply:
Makes sense to me, thanks!

/cc @rmgogogo 
as this might impact IR compliance to component spec

Reply:
IR will have component/PipelineNode level cache setting.